spec/lib/decidim/proposals/import/proposal_answer_creator_spec.rb

```diff
@@ -0,0 +1,166 @@
+      expect(record.id).to eq(data[:id])
+      expect(record.answer["en"]).to eq(data[:"answer/en"])
+      expect(record[:state]).to eq(data[:state])
+      expect(record.answered_at).to be >= (moment)
+    end
+
+    context "with an emendation" do
+      let!(:amendable) { create(:proposal, component: component) }
+      let!(:amendment) { create(:amendment, amendable: amendable, emendation: proposal, state: "evaluating") }
+
+      it "does not produce a record" do
+        record = subject.produce
+
+        expect(record).to be(nil)
+      end
+    end
+  end
+
+  describe "#finish!" do
+    it "saves the proposal" do
+      record = subject.produce
+      subject.finish!
+      expect(record.new_record?).to be(false)
+    end
+
+    it "creates an admin log record" do
+      record = subject.produce
+      subject.finish!
+
+      log = Decidim::ActionLog.last
+      expect(log.resource).to eq(record)
+      expect(log.action).to eq("answer")
+    end
+
+    context "when proposal state changes" do
+      context "when proposal had already a state" do
+        let!(:proposal) { create(:proposal, :evaluating, component: component) }
+        let(:state) { "accepted" }
+
+        it "returns broadcast :ok" do
+          expect(subject.finish!).to eq({ ok: [] })
+        end
+
+        context "and notifies followers" do
+          before do
+            allow(::Decidim::Proposals::Admin::NotifyProposalAnswer).to receive(:call).with(proposal, "evaluating")
+          end
+
+          it "notifies followers" do
+            subject.finish!
+            expect(::Decidim::Proposals::Admin::NotifyProposalAnswer).to have_received(:call)
+          end
+        end
+      end
+
+      context "when proposal had no state" do
+        let!(:proposal) { create(:proposal, :not_answered, component: component) }
+        let(:state) { "accepted" }
+
+        it "returns broadcast :ok" do
+          expect(subject.finish!).to eq({ ok: [] })
+        end
+
+        context "and notifies followers" do
+          before do
+            allow(::Decidim::Proposals::Admin::NotifyProposalAnswer).to receive(:call).with(proposal, "")
+          end
+
+          it "notifies followers" do
+            subject.finish!
+            expect(::Decidim::Proposals::Admin::NotifyProposalAnswer).to have_received(:call)
+          end
+        end
+      end
+
+      context "when proposal was just created and had a state set to nil" do
+        let!(:proposal) { create(:proposal, component: component) }
+        let(:state) { "accepted" }
+
+        before do
+          proposal.update(state: nil)
+        end
+
+        it "returns broadcast :ok" do
+          expect(subject.finish!).to eq({ ok: [] })
+        end
+
+        context "and notifies followers" do
+          before do
+            allow(::Decidim::Proposals::Admin::NotifyProposalAnswer).to receive(:call).with(proposal, "")
+          end
+
+          it "notifies followers" do
```

Comment:
Duplicate of line :120  
Please define a new shared_example "notifies followers"